src/stdlib_stats_distribution_uniform.fypp

```diff
@@ -0,0 +1,526 @@
+#:include "common.fypp"
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
+#:set ALL_KINDS_TYPES = INT_KINDS_TYPES + RC_KINDS_TYPES
+module stdlib_stats_distribution_uniform
+    use stdlib_kinds
+    use stdlib_error, only : error_stop
+    use stdlib_stats_distribution_PRNG, only : dist_rand
+
+    implicit none
+    private
+
+    real(dp), parameter  :: MESENNE_NUMBER = 1.0_dp / (2.0_dp ** 53 - 1.0_dp)
+    integer(int64), parameter :: INT_ONE = 1_int64
+
+    public :: rvs_uniform
+    public :: pdf_uniform
+    public :: cdf_uniform
+    public :: shuffle
+
+
+    interface rvs_uniform
+    !! Version experimental
+    !!
+    !! Get uniformly distributed random variate for integer, real and complex
+    !! variables.
+    !! ([Specification](../page/specs/stdlib_stats_distribution_uniform.html#
+    !! description))
+
+        module procedure unif_dist_rvs_0_rsp                 ! 0 dummy variable
```

Comment:
Suggest that you replace `unif_dist_rvs_` with `rvs_uniform_` where it occurs in function names, to make it easier for readers who will be looking for the latter.